client/clip_client/client.py

```diff
@@ -474,3 +486,262 @@ async def arank(self, docs: Iterable['Document'], **kwargs) -> 'DocumentArray':
+        parameters: Optional[Dict] = None,
+    ):
+        """Index the images or texts where their embeddings are computed by the server CLIP model.
+
+        Each image and text must be represented as a string. The following strings are acceptable:
+            - local image filepath, will be considered as an image
+            - remote image http/https, will be considered as an image
+            - a dataURI, will be considered as an image
+            - plain text, will be considered as a sentence
+        :param content: an iterator of image URIs or sentences, each element is an image or a text sentence as a string.
+        :param batch_size: the number of elements in each request when sending ``content``
+        :param show_progress: if set, show a progress bar
+        :param parameters: the parameters for the indexing, you can specify the model to use when you have multiple models
+        :return: the embedding in a numpy ndarray with shape ``[N, D]``. ``N`` is in the same length of ``content``
+        """
+        ...
+
+    @overload
+    def index(
+        self,
+        content: Union['DocumentArray', Iterable['Document']],
+        *,
+        batch_size: Optional[int] = None,
+        show_progress: bool = False,
+        parameters: Optional[dict] = None,
+    ) -> 'DocumentArray':
+        """Index the images or texts where their embeddings are computed by the server CLIP model.
+
+        :param content: an iterable of :class:`docarray.Document`, each Document must be filled with `.uri`, `.text` or `.blob`.
+        :param batch_size: the number of elements in each request when sending ``content``
+        :param show_progress: if set, show a progress bar
+        :param parameters: the parameters for the indexing, you can specify the model to use when you have multiple models
+        :return: the embedding in a numpy ndarray with shape ``[N, D]``. ``N`` is in the same length of ``content``
+        """
+        ...
+
+    def index(self, content, **kwargs):
+        if isinstance(content, str):
+            raise TypeError(
+                f'content must be an Iterable of [str, Document], try `.index(["{content}"])` instead'
+            )
+
+        self._prepare_streaming(
+            not kwargs.get('show_progress'),
+            total=len(content) if hasattr(content, '__len__') else None,
+        )
+        results = DocumentArray()
+        with self._pbar:
+            parameters = kwargs.pop('parameters', None)
+            self._client.post(
+                on='/index',
+                **self._get_post_payload(content, kwargs),
+                on_done=partial(self._gather_result, results=results),
+                parameters=parameters,
+            )
+
+        for c in content:
+            if hasattr(c, 'tags') and c.tags.pop('__loaded_by_CAS__', False):
+                c.pop('blob')
+
+        return self._unboxed_result(results)
+
+    @overload
+    async def aindex(
```

Comment:
why do we need `aindex` in the client? The logic of index in the server is not blocked since we use a queue

Reply:
See tips in this section ->https://clip-as-service.jina.ai/user-guides/client/#async-encoding